### ISSUE TITLE
InstanceNormalization: The parameter is incorrect error on DirectML

### ISSUE THREAD

**Issue by harishanand95** [microsoft/onnxruntime#12677]
**Describe the bug**
Running a group norm in the DirectML EP causes following error message.
```
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running InstanceNormalization node. Name:'InstanceNormalization_4' Status Message: D:\a\_work\1\s\onnxruntime\core\providers\dml\DmlExecutionProvider\src\MLOperatorAuthorImpl.cpp(1857)\onnxruntime_pybind11_state.pyd!00007FFF44011C2A: (caller: 00007FFF4401322F) Exception(3) tid(264) 80070057 The parameter is incorrect.
```

**Urgency**
None, used in stable diffusion repo. 

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- ONNX Runtime installed from (source or binary): onnxruntime-directml==1.12.1
- ONNX Runtime version: 1.12.1
- Python version: 3.9.12
- Visual Studio version (if applicable): 
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: RX 6800 XT

**To Reproduce**
- Describe steps/code to reproduce the behavior.
```python

import torch
import onnxruntime as ort

model = torch.nn.GroupNorm(num_groups=32, num_channels=320, eps=1e-6, affine=True)

check_inputs = [(torch.rand(1, 320, 320))]
model(*check_inputs[0])

traced_model = torch.jit.trace(model, (torch.rand(1, 320, 320)), strict=True)

torch.onnx.export(traced_model, check_inputs[0], "model.onnx", input_names=["input"], dynamic_axes={"input": [0]}, verbose=False)

so = ort.SessionOptions()
so.enable_mem_pattern=False
sess = ort.InferenceSession("model.onnx", so, providers=["DmlExecutionProvider"])

inputs = {"input": torch.rand(1, 320, 320).numpy(),}
noise_pred = sess.run(None, inputs)[0]
```

- Attach the ONNX model to the issue (where applicable) to expedite investigation.

**Expected behavior**
Execute similar to CPU EP.

**ONNX Model**
[model_onnx.zip](https://github.com/microsoft/onnxruntime/files/9399255/model_onnx.zip)

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by fdwr** [comment]
After turning on the D3D12 debug layer (Start / Run / dxcpl.exe), I see the following message:

```
D3D12 ERROR: Mismatched tensor sizes. Tensor 'Input' has size of 1 in dimension 1, while tensor 'Scale' has size of 32 in dimension 1. Both tensors are expected to have the same sizes in all dimensions. [ UNKNOWN ERROR #1: STRING_FROM_APPLICATION]
S:\WindowsAI\dml\SharedValidation/TensorValidator.h(439)\DirectML.Debug.dll!00007FF90212DE9B: (caller: 00007FF90211B969) Exception(1) tid(9af4) 80070057 The parameter is incorrect.
    [TensorValidator::ValidateIsBroadcastCompatibleWithFirst(E_INVALIDARG)]
```

It appears the DML EP's InstanceNormalization operator expects NCHW layout, whereas the input tensor here is reshaped to 3D. Until a fix can be made @harishanand95, consider unsqueezing the InstanceNormalization's input to include a placeholder  1 dimension (either in the batch count or the width dimension, depending on which axes you need).

**Comment by harishanand95** [comment]
Thanks @fdwr, here is my second attempt

```python
import torch
import onnxruntime as ort

model = torch.nn.GroupNorm(num_groups=320, num_channels=320, eps=1e-6, affine=True)

check_inputs = [(torch.rand(1, 320, 320, 1),)]
model(*check_inputs[0])

traced_model = torch.jit.trace(model, (torch.rand(1, 320, 320, 1),), strict=True)

torch.onnx.export(traced_model, check_inputs[0], "model.onnx", input_names=["input"], dynamic_axes={"input": [0]}, verbose=False)

so = ort.SessionOptions()
so.enable_mem_pattern=False
sess = ort.InferenceSession("model.onnx", so, providers=["DmlExecutionProvider"])

inputs = {"input": torch.rand(1, 320, 320, 1).numpy(),}
noise_pred = sess.run(None, inputs)[0]
```
Tried this, still failing with an error. Works on CPU EP. I tried setting debug layer, it did not work either  :-/
```
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running InstanceNormalization node. Name:'InstanceNormalization_4' Status Message: D:\a\_work\1\s\onnxruntime\core\providers\dml\DmlExecutionProvider\src\MLOperatorAuthorImpl.cpp(1857)\onnxruntime_pybind11_state.pyd!00007FFD13141C2A: (caller: 00007FFD1314322F) Exception(3) tid(3558) 80070057 The parameter is incorrect.
```

**Comment by fdwr** [comment]
> I tried setting debug layer, it did not work either :-/

The debug layer is for extra message output. Click "Edit list", add "Python.exe", and then set "Force on" for the debug layer.

> Tried this, still failing with an error.

Well once https://github.com/microsoft/onnxruntime/pull/12693 finishes (there are fluke failures in the CI test), then your earlier model should work.

**Comment by fdwr** [comment]
https://github.com/microsoft/onnxruntime/pull/12693 is complete. 3D tensors work for InstanceNormalization.

**Comment by harishanand95** [comment]
Thank you @fdwr, I tried using the latest nightly onnxruntime directml and it still fails on this issue,  

Got the model from this link https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly-directml/overview/1.13.0.dev20220823003

```bash
$ pip freeze | grep direct
ort-nightly-directml @ file:///C:/Users/haris/Desktop/harishanand95_diff/ort_nightly_directml-1.13.0.dev20220823003-cp38-cp38-win_amd64.whl
```

After building from the latest main, the above code works.

Any ideas where I can the latest *.whl or nightly package files? When are the onnxruntime-direct releases? Thanks!!

cc: @simona198710, in case you are also facing the same issues

**Comment by fdwr** [comment]
@harishanand95 Try the nightly package from the 30th? https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly-directml/overview/1.13.0.dev20220830001 (the package from the 23rd wouldn't include may change because I completed the change August 24th). Glad to hear the one built from main works :).

**Comment by harishanand95** [comment]
Thanks! the latest package works!

**Comment by pauldog** [comment]
> @harishanand95 Try the nightly package from the 30th? https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly-directml/overview/1.13.0.dev20220830001 (the package from the 23rd wouldn't include my change because I completed the change August 24th). Glad to hear the one built from main works :).

Hi is this version is 1.13.0 but I am getting a similar error to the OP but I have downloaded the nuget package for 1.13.1 

So does that mean it's a different error?

**Comment by harishanand95** [comment]
Hey @pauldog, I'm not sure how to fix that. If you are attempting stable diffusion on AMD GPU, you can try our new approach which is definitely faster. 
https://github.com/nod-ai/SHARK/blob/main/shark/examples/shark_inference/stable_diffusion/stable_diffusion_amd.md

**Comment by pauldog** [comment]
> Hey @pauldog, I'm not sure how to fix that. If you are attempting stable diffusion on AMD GPU, you can try our new approach which is definitely faster. https://github.com/nod-ai/SHARK/blob/main/shark/examples/shark_inference/stable_diffusion/stable_diffusion_amd.md

Thanks I'll check it out.

**Comment by patsch007** [comment]
Hi @harishanand95 have you checked what directml.dll you use. 
I was running today into a similar problem. It was the wrong directml.dll from the System32 loaded.
As quick fix i just putted the latest version of directml in the folder of the application. 

In the latest Windows 11 (Build 10.0.22621; 22H2) is still only directml 1.8.0 included but you need 1.10.0 for onnx version 1.13.0 - 1.14.0. 
More information you find here https://learn.microsoft.com/en-us/windows/ai/directml/dml-version-history
